### PR TITLE
Update Cart mutations to use new API

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Update to use new API for managing the cart

## Justify why they are needed

Get rid of "cartId"

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/components/ProductPage/PurchaseForm/useHandleSubmitAddToCart.ts [hunk 1]
```diff
@@ -77,12 +78,17 @@ const useCartEntryAdd = (mutationOptions: CartEntryAddOptions = {}) => {
...
     (options: CartEntryAddOptions = {}) => {
       const handleCompleted = (data: CartEntryAddMutation) => {
-        if (!data.cartEntriesAdd.cart) return
+        const updatedShopSession = data.shopSessionCartEntriesAdd.shopSession
+        console.log('updatedShopSession', updatedShopSession)
+        if (!updatedShopSession) return
 
         const { variables } = options
-        const addedOffer = data.cartEntriesAdd.cart.entries.find(
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```
Comment (robinandeer) [inline apps/store/src/components/ProductPage/PurchaseForm/useHandleSubmitAddToCart.ts:86]: couldn't figure out why this is causing a listing error

Comment (alebedev) [reply apps/store/src/components/ProductPage/PurchaseForm/useHandleSubmitAddToCart.ts:84]: Can you double-check in CLI?  Should not fail

File: apps/store/src/graphql/CartUnredeemCampaign.graphql [hunk 2]
```diff
@@ -1,5 +1,12 @@
-mutation UnredeemCampaign($cartId: UUID!, $campaignId: String!) {
-  cartUnredeemCampaign(input: { cartId: $cartId, campaignId: $campaignId }) {
-    ...CartFragment
+mutation UnredeemCampaign($shopSessionId: UUID!, $campaignId: String!) {
+  shopSessionCartCampaignUnredeem(
+    input: { shopSessionId: $shopSessionId, campaignId: $campaignId }
+  ) {
+    shopSession {
```
Comment (alebedev) [inline apps/store/src/graphql/CartUnredeemCampaign.graphql:5]: Nit: maybe we want to declare ShopSessionMutationOutput and reuse it?

Comment (alebedev) [inline apps/store/src/graphql/CartUnredeemCampaign.graphql:5]: Nit: We only have 1 use of CartFragment remaining, what do you think about inlining it?

**Review by alebedev — APPROVED** [review]
Looks good!

What do you think about using context for shopSessionId instead of prop drilling?

**Pull request merged**